WalletWasabi/Wallets/Wallet.cs

```diff
@@ -144,16 +144,12 @@ public HdPubKey GetNextReceiveAddress(IEnumerable<string> destinationLabels)
 		return KeyManager.GetNextReceiveKey(new LabelsArray(destinationLabels));
 	}
 
-	private double GetPrivacyPercentage(CoinsView coins, int privateThreshold)
+	public int GetPrivacyPercentage()
 	{
-		var privateAmount = coins.FilterBy(x => x.IsPrivate(privateThreshold)).TotalAmount();
-		var normalAmount = coins.FilterBy(x => !x.IsPrivate(privateThreshold)).TotalAmount();
+		var currentPrivacyScore = Coins.Sum(x => x.Amount.Satoshi * Math.Min(x.HdPubKey.AnonymitySet - 1, x.IsPrivate(AnonScoreTarget) ? AnonScoreTarget - 1 : AnonScoreTarget - 2));
+		var maxPrivacyScore = Coins.TotalAmount().Satoshi * (AnonScoreTarget - 1);
+		int pcPrivate = maxPrivacyScore == 0M ? 100 : (int)(currentPrivacyScore * 100 / maxPrivacyScore);
```

Comment:
I just don't get it how the `maxPrivacyScore` will ever be `0M`. 
Unless `maxPrivacyScore == 0M ? 100` part is only responsible for empty wallets.

Is that the case?

Reply:
Yes.